### PR TITLE
Remove Search Registries for Builds

### PR DESCRIPTION
**Pull request by adambkaplan**
The discussion in https://github.com/openshift/builder/pull/40 made it clear that registry search is incompatible with the auth mechanisms we have in place. This was the case when we were using docker, and is even more true now that buildah is driving builds in 4.0. Allowing customers to have anything except `docker.io` on the search path leads to potential security leaks, as credentials may be sent to the wrong repository if the pull spec does not have a fully qualified image name. Furthermore, removing `docker.io` from the search path causes builds to break, as buildah uses `docker.io` in its search path for image pull, but not `bud` builds (see https://github.com/containers/buildah/issues/1316).

This PR removes the ability to configure registry search for builds. Builds will default to `docker.io` - and only `docker.io` - for search.

### CODE REVIEW

**Comment by adambkaplan** [comment]
/hold

Need PM input.

**Comment by bparees** [comment]
@adambkaplan go ahead and line up the origin PR that rips out the rest of this so once this merges we can minimize the window during which origin is not bumpable.

(this should have no impact on client-go so i don't think a PR there is needed)

**Comment by adambkaplan** [comment]
/hold cancel

PM approval from @mpbarrett and @sspeiche

**Comment by bparees** [comment]
/lgtm
as this is dropping a field that never shipped i don't see there being any controversy.

**Comment by bparees** [comment]
/hold
re-holding until the origin PR merges so we don't get in a state where we can't bump origin.

File: config/v1/types_build.go [hunk 1]
```diff
@@ -78,12 +78,6 @@ type ImageLabel struct {
 }
 
 type RegistriesConfig struct {
```
[the remark this answers is not in the document]
Comment (bparees) [reply config/v1/types_build.go]: do we know we will always want the same set of registries to be blocked/allowed for both builds+runtime?

[the remark this answers is not in the document]
Comment (mrunalp) [reply config/v1/types_build.go]: Can't think of use cases where we wouldn't have the same set for both.

**Comment by adambkaplan** [comment]
/hold cancel

openshift/origin#21939 landed, so this won't block origin.

Note that this will be replaced by a global config in https://github.com/openshift/api/pull/204

**Comment by adambkaplan** [comment]
Note that with #204 this whole thing will eventually go away.

**Pull request merged**